app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/ConfirmYourDecisionController.scala

```diff
@@ -0,0 +1,86 @@
+ */
+
+package uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future.successful
+
+import play.api.data.Form
+import play.api.data.Forms._
+import play.api.mvc.{MessagesControllerComponents, _}
+import uk.gov.hmrc.apiplatform.modules.common.services.ApplicationLogger
+import uk.gov.hmrc.apiplatform.modules.stride.config.StrideAuthConfig
+import uk.gov.hmrc.apiplatform.modules.stride.connectors.AuthConnector
+import uk.gov.hmrc.apiplatform.modules.stride.controllers.GatekeeperBaseController
+import uk.gov.hmrc.apiplatform.modules.stride.controllers.actions.ForbiddenHandler
+import uk.gov.hmrc.apiplatform.modules.submissions.services.SubmissionService
+
+import uk.gov.hmrc.apigatekeeperapprovalsfrontend.config.ErrorHandler
+import uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.actions.ApplicationActions
+import uk.gov.hmrc.apigatekeeperapprovalsfrontend.domain.models.ApplicationId
+import uk.gov.hmrc.apigatekeeperapprovalsfrontend.services.ApplicationActionService
+import uk.gov.hmrc.apigatekeeperapprovalsfrontend.views.html.{ApplicationApprovedPage, ApplicationDeclinedPage, ConfirmYourDecisionPage}
+
+object ConfirmYourDecisionController {
+  case class ViewModel(applicationId: ApplicationId, appName: String)
+
+  case class DeclineForm(reasons: String)
+
+  val declineForm: Form[DeclineForm] = Form(
+    mapping(
+      "reasons" -> nonEmptyText
+    )(DeclineForm.apply)(DeclineForm.unapply)
+  )
+}
+
+@Singleton
+class ConfirmYourDecisionController @Inject()(
+  strideAuthConfig: StrideAuthConfig,
+  authConnector: AuthConnector,
+  forbiddenHandler: ForbiddenHandler,
+  mcc: MessagesControllerComponents,
+  val errorHandler: ErrorHandler,
+  val applicationActionService: ApplicationActionService,
+  val submissionService: SubmissionService,
+  confirmYourDecisionPage: ConfirmYourDecisionPage,
+  applicationApprovedPage: ApplicationApprovedPage,
+  applicationDeclinedPage: ApplicationDeclinedPage
+)(implicit override val ec: ExecutionContext) 
+    extends GatekeeperBaseController(strideAuthConfig, authConnector, forbiddenHandler, mcc)
+    with ApplicationActions
+    with ApplicationLogger {
+  import ConfirmYourDecisionController._
+
+  def page(applicationId: ApplicationId): Action[AnyContent] = loggedInWithApplicationAndSubmission(applicationId) { implicit request =>
+    successful(Ok(confirmYourDecisionPage(ViewModel(applicationId, request.application.name))))
+  }
+
+  def action(applicationId: ApplicationId): Action[AnyContent] = loggedInWithApplicationAndSubmission(applicationId) { implicit request => 
+    request.body.asFormUrlEncoded.getOrElse(Map.empty).get("grant-decision").flatMap(_.headOption) match {
+      case Some("decline")              => successful(Redirect(uk.gov.hmrc.apigatekeeperapprovalsfrontend.controllers.routes.DeclinedJourneyController.provideReasonsPage(applicationId)))
+      // TODO
```

Comment:
is this needed?

Reply:
Yes, in so much as we need to play the next ticket for this.